packages/v4/src/content/design-guidelines/components/modal/modal.md

```diff
@@ -7,111 +7,131 @@ related: ['Button']
 
 ## Usage
-### When to use
-Use a modal dialog when information is critical and requires user input or attention.
-Use modals to:
 
-* [Validate user decisions](#validate-user-decisions)
-* [Report errors](#report-errors)
-* [Provide the result of a task or state of an application](#provide-the-result-of-a-task-or-state-of-an-application)
-* [Get required user input](#get-required-user-input)
+Use a modal to communicate critical information that requires user input or action. 
+
+Modals can:
+
+* Validate user decisions.
+* Confirm non-destructive or destructive actions.
+* Report errors.
+* Share task results and statuses.
+* Prompt required user input.
+
+Don’t use a modal unless its information is vital to user success and worth disrupting a user’s task flow.
+
+## Types of modals
+
+Modals serve several functions in a UI and appear in three main types:
+
+* [Confirmation dialogs](#confirmation-dialogs)
+
+* [Error dialogs](#error-dialogs)
+
+* [Passive dialogs](#passive-dialogs)
+
+### Confirmation dialogs
+
+Use a confirmation dialog to validate user decisions and communicate their consequences. Confirmation dialogs can confirm non-destructive and destructive actions.
+
+#### Confirm a non-destructive action
+
+<img src="./img/non-destructive-confirmation-dialog.png" alt="A side-by-side comparison of unsuccessful and successful confirmaton dialogs for leaving a page without saving. The successful dialog explains the consequence of this action: All unsaved informaton will be lost." />
+
+When confirming a non-destructive action, provide specific context about what will happen once a user confirms it.
+
+#### Confirm a destructive action
+
+<img src="./img/destructive-confirmation-dialog.png" alt="A side-by-side comparison of unsuccessful and successful destructive confirmaton dialogs for deleting a record. The successful dialog explains the consequence of deleting this record: Record name, location, and notes will be lost." />
+
+Use a [primary button](/components/button/design-guidelines#primary-button) to confirm a destructive action. If the action carries serious consequences, then use a [danger button](/components/button/design-guidelines#danger-button) instead.
+
+If an action is difficult or impossible to undo, add a warning icon beside the headline. 
+
+If a destructive action has serious consequences, consider adding an additional step before its destructive button becomes active. Additional confirmation measures like [text input](/components/text-input/design-guidelines) fields safeguard a user from accidentally completing the action. 
 
-### When not to use
-Modal dialogs interrupt user task flow and should be avoided for any information that is not critical to user success.
+When a text input field is added to a confirmation dialog, the danger button will only become selectable once a user enters the entire word or phrase.
```

Comment:
I wonder if saying "the danger button will only become enable once a user..." instead of selectable would make more sense, to highlight that it goes from literally being disabled to enabled. But just a suggestion, will leave final call to you!